css-highlight-api-1/Overview.bs

```diff
@@ -680,24 +681,24 @@ method must return the result of running these steps:
 	* The topmost [=box=] in the [=viewport=] in paint order that would be a target for hit testing at coordinates <var>x</var>,<var>y</var> when applying
 		the [=transforms=] that apply to the descendants of the viewport, has an element associated to it that's in a [=shadow tree=] whose
 		[=shadow root=] is not [=list/contains|contained by=] <var>options</var>.<var>shadowRoots</var>.
-1. Otherwise, return a [=sequence=] of {{HighlightHitResult}} objects ordered in [=priority|reverse paint order=] of their <var>highlight</var> attributes,
-	which are contained in this {{HighlightRegistry}}.
-	Every {{HighlightHitResult}} <var>highlightHitResult</var> object in the returned sequence has a non-empty <var ignore=''>ranges</var> attribute,
-	ordered in the same way as [=custom highlights=] [=set entries=], where each {{AbstractRange}} <var>abstractRange</var> contained in it satisfies the following:
-
-		* <var>abstractRange</var> is in <var>highlightHitResult</var>.<var>highlight</var> [=set entries=].
-
-		* <var>abstractRange</var> is either a live {{Range}}, or a [=StaticRange/valid=] {{StaticRange}} that allows to successfully construct a {{Range}} object <var>range</var> 
-		    whose [=start node=] and [=end node=] are set to <var>abstractRange</var>'s [=start node=] and [=end node=] respectively,
-			and [=start offset=] and [=end offset=] are set to <var>abstractRange</var>'s [=start offset=] and [=end offset=] respectively.
-
-		* The coordinates <var>x</var>,<var>y</var> fall inside at least one of the {{DOMRect}}s returned by calling {{Range/getClientRects()}} on <var>range</var>.
+1. Otherwise, let <var>results</var> be an empty [=sequence=].
+1. For each {{Highlight}} <var>highlight</var> in this {{HighlightRegistry}}:
+	1. Create a {{HighlightHitResult}} <var>result</var> with the given <var>highlight</var> as <var>result</var>.{{HighlightHitResult/highlight}} attribute.
+	1. For each {{AbstractRange}} <var>abstractRange</var> in <var>highlight</var>:
+		1. If <var>abstractRange</var> is a live {{Range}} or a [=StaticRange/valid=] {{StaticRange}}, construct a {{Range}} object <var>range</var> 
```

Comment:
It's not really clear what the algorithm is supposed to do if abstractRang is not a live range or a valid staticrange. I think we need to [`continue`](https://infra.spec.whatwg.org/#iteration-continue) if it's not, or add another layer of nesting after this `if`.

Reply:
It might also be more understandable to check the condition that `abstractRange` is an invalid staticrange.